### PR TITLE
New module: monitoring/grafana_plugin

### PR DESCRIPTION
**Pull request by seuf**
##### SUMMARY
**grafana_plugin** is a new ansible module that allow you to install or remove grafana plugins.

##### ISSUE TYPE
 - New Module Pull Request

##### COMPONENT NAME
lib/ansible/modules/monitoring/grafana_plugin.py

##### ANSIBLE VERSION
```
ansible 2.3.1.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides
  python version = 2.7.13 (default, May 10 2017, 20:04:28) [GCC 6.3.1 20161221 (Red Hat 6.3.1-1)]
```

##### ADDITIONAL INFORMATION
This new module allow grafana plugins install and uninstall :
```
- hosts: grafana
  tasks:
  - name: install - update grafana piechart panel plugin
    grafana_plugin:
      name: grafana-piechart-panel
      version: latest
      state: present
```

### CODE REVIEW

File: lib/ansible/modules/monitoring/grafana_plugin.py [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+
+DOCUMENTATION = '''
+---
+module: grafana_plugin
+author:
+  - "Thierry Sallé (@tsalle)"
+version_added: "2.5"
+short_description: Manage grafana plugins via grafana-cli
+description:
+  - install / remove grafana plugins
```
Comment (gundalow) [inline lib/ansible/modules/monitoring/grafana_plugin.py]: `Install and remove grafana plugins.`

File: lib/ansible/modules/monitoring/grafana_plugin.py [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+  version:
+    description:
+      - Version of the plugin to install.
+      - Default to latest
+  grafana_plugins_dir:
+    description:
+      - Directory where grafana plugin will be installed.
+  grafana_repo:
+    description:
+      - Grafana repository ("https://grafana.net/api/plugins")
```
Comment (gundalow) [inline lib/ansible/modules/monitoring/grafana_plugin.py]: Should `https://grafana.net/api/plugins` be the `default:` (and in argspec)?`

**Review by gundalow — CHANGES_REQUESTED** [review]
Same comments as in https://github.com/ansible/ansible/pull/31930 apply here

File: lib/ansible/modules/monitoring/grafana_plugin.py (hunk 2 above)
Comment (seuf) [reply lib/ansible/modules/monitoring/grafana_plugin.py]: No, this option is only used when we want to use a custom grafana repository. (cf. https://github.com/grafana/grafana/pull/9299).
I've updated the description..

**Review by gundalow — APPROVED** [review]

**Comment by gundalow** [comment]
@seuf I made a few changes directly to this PR

Merged into `devel` will be released in Ansible 2.5, Thanks

**Comment by nerzhul** [comment]
nice job

**Pull request merged**